gochart.go

```diff
@@ -202,38 +184,19 @@ func (r *GoChart) ClearTextRotation() {
 
 // Save writes the image to the given writer.
 func (r *GoChart) Save(w io.Writer) error {
-	width, height := r.c.Size()
-	switch r.output {
-	case OutputSVG:
-		svg := NewSVG(w, width, height)
-		r.c.Render(svg)
-		return svg.Close()
-	case OutputPDF:
-		pdf := NewPDF(w, width, height)
-		r.c.Render(pdf)
-		return pdf.Close()
-	case OutputEPS:
-		eps := NewEPS(w, width, height)
-		r.c.Render(eps)
-		return nil
-	case OutputPNG:
-		img := r.c.WriteImage(1.0)
-		if err := png.Encode(w, img); err != nil {
-			return err
-		}
-		return nil
-	case OutputJPG:
-		img := r.c.WriteImage(1.0)
-		if err := jpeg.Encode(w, img, nil); err != nil {
-			return err
-		}
-		return nil
-	case OutputGIF:
-		img := r.c.WriteImage(1.0)
-		if err := gif.Encode(w, img, nil); err != nil {
-			return err
-		}
-		return nil
-	}
-	return fmt.Errorf("unknown output format")
+	return r.writer(w, r.c)
```

Comment:
This is a great improvement, the `Output` type was really an ugly temporary measure.